src/lib/y2storage/devicegraph.rb

```diff
@@ -350,27 +351,47 @@ def find_by_name(name)
     # * It avoids system lookup in potentially risky scenarios (like an outdated
     #   {StorageManager#probed}).
     #
-    # In case of LUKSes, the device might be found by using an alternative name,
+    # In case of LUKSes and MDs, the device might be found by using an alternative name,
     # see {#alternative_names}.
     #
-    # @see #deep_find
+    # @see #find_by_devicegraph_name
+    # @see #find_via_system_lookup
     #
-    # @param device_name [String] can be a kernel name like "/dev/sda1" or any symbolic
+    # @param dev_name [String] can be a kernel name like "/dev/sda1" or any symbolic
     #   link below the /dev directory
     # @param alternative_names [Boolean] whether to try the search with possible alternative names
     # @return [Device, nil] the found device, nil if no device matches the name
-    def find_by_any_name(device_name, alternative_names: true)
-      names = [device_name]
-      names = names.concat(alternative_names(device_name)) if alternative_names
+    def find_by_any_name(dev_name, alternative_names: true)
+      names = [dev_name]
+      names = names.concat(alternative_names(dev_name)) if alternative_names
 
-      device = nil
+      names.each do |name|
```

Comment:
Maybe it would be clearer with another method name, e.g., `#find_first_name`:

``` 
find_first_name(names) { |n| find_by_devicegraph_name(n) }

find_first_name(names) { |n| find_by_system_lookup(n) }
```